AllASPFuncs.py

```diff
@@ -70,107 +70,134 @@
 
-        spacing = 1.0
-        x = np.arange(difference, subsize+difference, spacing)
-        y = np.arange(difference, subsize+difference, spacing)
+    Inputs:
+    ra_center, dec_center: floats, coordinate center of the image
+    wcs: the WCS of the image, currently a galsim.fitswcs.AstropyWCS object
+    image: 2D numpy array of floats of shape (size x size), the image to build
+    the grid on.
+    percentiles: list of floats, the percentiles to use to bin the image. The
+                more bins, the more possible grid points could be placed in
+                that pixel. For instance, say if you had bins [45, 90],
+                as is default. A pixel that lies in the 30th percentile for
+                brightness would get 1 point. A pixel at the 50th percentiile
+                for brightness would get a 2x2 grid of points.
+                A pixel above the 90th percentile would get a 3x3 grid of
+                points. If you have more bins, you could go even higher to
+                4x4 and 5x5 etc. These points are evenly paced within the
+                pixel.
+    subsize: If you don't care as much about the edges of the image, you can
+             specify a maximum size in pixels for the grid to be. For instance
+             I could have an image that is 11x11 but a grid that is only 9x9.
+             This is useful and different from making a smaller image because
+             when the image rotates, model points near the corners of the image
+             may be rotated out. By taking a smaller grid, we can avoid this.
+             Int, width of the grid in pixels.
 
-        x -= np.mean(x)
-        x+= x_center
+    Returns:
+    ra_grid, dec_grid: 1D numpy arrays of floats, the RA and DEC of the grid.
+    '''
+    size = np.shape(image)[0]
+    Lager.debug('image shape: {}'.format(np.shape(image)))
+    Lager.debug('GRID TYPE: ADAPTIVE')
+    # Bin the image in logspace and allocate grid points based on the
+    # brightness.
 
-        y -= np.mean(y)
-        y+= y_center
+    difference = int((size - subsize)/2)
+    x_center, y_center = wcs.toImage(ra_center, dec_center, units='deg')
+    x = difference + np.arange(0, subsize, 1)
+    y = difference + np.arange(0, subsize, 1)
+
+    imcopy = np.copy(image)
+    # We need to make sure that the image is not zero, otherwise we get
+    # infinities in the log space.
+    imcopy[imcopy <= 0] = 1e-10
+    imcopy = np.log(imcopy)
+    bins = [0]
+    bins.extend(np.nanpercentile(imcopy, percentiles))
+    bins.append(100)
+    Lager.debug(f'BINS: {bins}')
+
+    a = np.digitize(imcopy, bins)
+    xes = []
+    ys = []
+
+    yvals = np.rint(y).astype(int)
+    xvals = np.rint(x).astype(int)
+    for xindex in xvals:
+        x = xindex + 1
+        for yindex in yvals:
+            y = yindex + 1
+            # xindex and yindex are the indices within the numpy array, while
+            # x and y are the actual locations in pixel space.
```

Comment:
You're carrying around this default assumption of how to refer to x, y coordinates in pixel space.  There are different conventions.  Three reasonable conventions of the center of the "first" pixel being defined at (0, 0), (0.5, 0.5), (1, 1)).  I think you think that one of them is "right".  But that's just one choice.  Be explicit about your assumption/choice.